.dotnet/tests/Batch/BatchTests.cs

```diff
@@ -22,14 +23,14 @@ public BatchTests(bool isAsync) : base(isAsync)
     }
 
     [Test]
-    public async Task ListBatchesProtocol()
+    public void ListBatchesProtocol()
     {
         BatchClient client = GetTestClient();
-        ClientResult result = IsAsync
-            ? await client.GetBatchesAsync(after: null, limit: null, options: null)
-            : client.GetBatches(after: null, limit: null, options: null);
+        var enumerator = client.GetBatches(after: null, limit: null, options: null).GetEnumerator();
 
-        BinaryData response = result.GetRawResponse().Content;
+        Assert.IsTrue(enumerator.MoveNext());
+
+        BinaryData response = enumerator.Current.GetRawResponse().Content;
```

Comment:
Would it be possible to illustrate what it looks like parsing the pages out of the collection of ClientResults in a foreach loop in these tests?